src/main/java/io/lakefs/PathMapper.java

```diff
@@ -0,0 +1,311 @@
+package io.lakefs;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * A class that does path mapping between configured pairs of path styles, and matches the appropriate path mapping for
+ * a converted path.
+ */
+public class PathMapper {
+
+    public static final Logger LOG = LoggerFactory.getLogger(PathMapper.class);
+
+    protected static final String MAPPING_CONFIG_PREFIX = "routerfs.mapping.";
+    protected static final String NON_DEFAULT_MAPPING_CONFIG_PATTERN = "^routerfs\\.mapping\\.([a-z0-9-_]*)\\.(\\d*)\\.(replace|with)";
```

Comment:
```suggestion
    protected static final String NON_DEFAULT_MAPPING_CONFIG_PATTERN = "^routerfs\\.mapping\\.([-a-z0-9_]*)\\.(\\d*)\\.(replace|with)";
```
Placing a dash inside a character group is always confusing because it _might_ be a range.  The only place that is entirely clear is right at the start.

Reply:
Thanks! changed it